submultimi/submultimi.cpp

```diff
@@ -0,0 +1,31 @@
+//https://infoarena.ro/problema/submultimi
+
+#include <fstream>
+#include <vector>
+
+using namespace std;
+ifstream fin ("submultimi.in");
+ofstream fout ("submultimi.out");
+
+vector <int> submultime;
+
+void solution(int position, int N, int lastValue) {
+    for (int i = lastValue; i <= N; ++i) {
+        submultime.push_back(i);
+        for (int j = 0; j < submultime.size(); ++j) {
+            fout << submultime[j] << " ";
+        }
+        fout << "\n";
+        lastValue += 1;
+        solution(position + 1, N, lastValue);
```

Comment:
This could be `solution(position + 1, N, i + 1)`. This way you dont need to keep incrementing lastValue.